subsys/net/lib/sntp/sntp.c

```diff
@@ -0,0 +1,261 @@
+
+#define SNTP_LI_MAX 3
+#define SNTP_VERSION_NUMBER 3
+#define SNTP_MODE_CLIENT 3
+#define SNTP_MODE_SERVER 4
+#define SNTP_STRATUM_KOD 0 /* kiss-o'-death */
+#define OFFSET_1970_JAN_1 2208988800
+
+#define SNTP_CTX_SRV_SOCKADDR(ctx) (ctx->net_app_ctx.default_ctx->remote)
+
+static void sntp_pkt_dump(struct sntp_pkt *pkt)
+{
+	if (pkt) {
+		NET_DBG("li               %x", pkt->li);
+		NET_DBG("vn               %x", pkt->vn);
+		NET_DBG("mode             %x", pkt->mode);
+		NET_DBG("stratum:         %x", pkt->stratum);
+		NET_DBG("poll:            %x", pkt->poll);
+		NET_DBG("precision:       %x", pkt->precision);
+		NET_DBG("root_delay:      %x", pkt->root_delay);
+		NET_DBG("root_dispersion: %x", pkt->root_dispersion);
+		NET_DBG("ref_id:          %x", pkt->ref_id);
+		NET_DBG("ref_tm_s:        %x", pkt->ref_tm_s);
+		NET_DBG("ref_tm_f:        %x", pkt->ref_tm_f);
+		NET_DBG("orig_tm_s:       %x", pkt->orig_tm_s);
+		NET_DBG("orig_tm_f:       %x", pkt->orig_tm_f);
+		NET_DBG("rx_tm_s:         %x", pkt->rx_tm_s);
+		NET_DBG("rx_tm_f:         %x", pkt->rx_tm_f);
+		NET_DBG("tx_tm_s:         %x", pkt->tx_tm_s);
+		NET_DBG("tx_tm_f:         %x", pkt->tx_tm_f);
+	}
+}
+
+static s32_t parse_response(u8_t *data, u16_t len, u32_t orig_ts,
+				u64_t *epoch_time)
+{
+	struct sntp_pkt *pkt = (struct sntp_pkt *)data;
+
+	sntp_pkt_dump(pkt);
+
+	if (ntohl(pkt->orig_tm_s) != orig_ts) {
+		NET_DBG("Unexpected originate timestamp");
+		return -EINVAL;
+	}
+
+	if (pkt->li > SNTP_LI_MAX) {
+		NET_DBG("Unexpected LI");
+		return -EINVAL;
+	}
+
+	if (pkt->mode != SNTP_MODE_SERVER) {
+		/* For unicast and manycast, server should return 4.
+		 * For broadcast (which is not supported now), server should
+		 * return 5.
+		 */
+		NET_DBG("Unexpected mode");
+		return -EINVAL;
+	}
+
+	if (pkt->stratum == SNTP_STRATUM_KOD) {
+		NET_DBG("kiss-o'-death stratum");
+		return -EBUSY;
+	}
+
+	if (epoch_time) {
+		u64_t ts = (u64_t)ntohl(pkt->tx_tm_s);
+
+		/* Check if most significant bit is set */
+		if ((ts >> 31) != 0) {
+			/* UTC time is reckoned from 0h 0m 0s UTC
+			 * on 1 January 1900.
+			 */
+			if (ts >= OFFSET_1970_JAN_1) {
+				*epoch_time = ts - OFFSET_1970_JAN_1;
+			} else {
+				return -EINVAL;
+			}
+		} else {
+			/* UTC time is reckoned from 6h 28m 16s UTC
+			 * on 7 February 2036.
+			 */
+			*epoch_time = ts + 0x100000000 - OFFSET_1970_JAN_1;
+		}
+	}
+
+	return 0;
+}
+
+static void sntp_recv_cb(struct net_app_ctx *ctx, struct net_pkt *pkt,
+			int status, void *user_data)
+{
+	struct sntp_ctx *sntp = NULL;
+
+	if (!pkt) {
+		return;
+	}
+
+	if (status < 0) {
+		return;
+	}
+
+	if (!user_data) {
+		goto error_exit;
+	}
+	sntp = (struct sntp_ctx *)user_data;
+
+	if (net_pkt_appdatalen(pkt) != sizeof(struct sntp_pkt)) {
+		goto error_exit;
+	}
+
+	if (!net_pkt_appdata(pkt)) {
+		goto error_exit;
+	}
+
+	k_fifo_put(&sntp->recv_q, pkt);
+	return;
+
+error_exit:
+	if (ctx) {
+		k_fifo_cancel_wait(&sntp->recv_q);
+	}
+	net_pkt_unref(pkt);
```

Comment:
always break a line after a } (unless it's another } etc..)

Reply:
This commit looks like it could be merged to some earlier commit in your PR. I mean that you are fixing here something that you introduced in this PR.